src/Diagrams/Swimunit/Dotmatrix.hs

```diff
@@ -0,0 +1,843 @@
+{-# LANGUAGE NoMonomorphismRestriction #-}
+
+module Diagrams.Swimunit.Dotmatrix
+where
+
+import qualified Data.Map as Map
+
+import Diagrams.Prelude
+import Diagrams.Backend.SVG.CmdLine
```

Comment:
That's fine. I'd say replace all the `Diagram B` with `QDiagram b V2 n Any` and `ghc` should tell you what to do if there's any errors. If you're still getting stuck you can ask here or on irc.